controllers/placementapi_controller.go

```diff
@@ -766,7 +766,17 @@ func (r *PlacementAPIReconciler) reconcileNormal(ctx context.Context, instance *
 
 	if instance.Status.ReadyCount > 0 {
 		instance.Status.Conditions.MarkTrue(condition.DeploymentReadyCondition, condition.DeploymentReadyMessage)
+	} else {
+		Log.Info("Deployment is not ready")
+		instance.Status.Conditions.Set(condition.FalseCondition(
+			condition.DeploymentReadyCondition,
+			condition.RequestedReason,
+			condition.SeverityInfo,
+			condition.DeploymentReadyRunningMessage))
+		// It is OK to return success as we are watching for StatefulSet changes
+		return ctrl.Result{}, nil
```

Comment:
What if `replicas` is set to `0`?  That might prevent the CR from ever reaching the "ready" state [1].

[1] https://github.com/openstack-k8s-operators/placement-operator/blob/4daf37ecb2bb584e1f3a98b7f1660b81519f8ff1/api/v1beta1/placementapi_types.go#L196-L199

Reply:
done